File: src/main/java/io/tatum/model/response/ledger/AccountBalance.java
package io.tatum.model.response.ledger;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

/**
 * The type Account balance.
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode
@JsonIgnoreProperties(ignoreUnknown = true)
public class AccountBalance {
    private BigDecimal accountBalance;
    private BigDecimal availableBalance;
}

File: src/main/kotlin/io/tatum/model/response/ledger/AccountBalance.kt
package io.tatum.model.response.ledger

/**
 * @property accountBalance Account balance represents all assets on the account, available and blocked.
 * @property availableBalance Available balance on the account represents account balance minus blocked amount on the account.
 */
class AccountBalance(
    val accountBalance: String,
    val availableBalance: String,
)
